packages/augur-state/src/db/DB.ts

```diff
@@ -103,43 +120,51 @@ export class DB<TBigNumber> {
   /**
    * Syncs generic events and user-specific events with blockchain and updates MetaDB info.
    *
-   * @param {Augur<TBigNumber>} augur Augur object with which to sync
+   * TODO Make use of EventLogDBRouter
+   *
+   * @param {Provider} ethereumProvider Ethereum provider to use for syncing
+   * @param {Events} events Object used for getting and parsing logs from the blockchain
    * @param {number} chunkSize Number of blocks to retrieve at a time when syncing logs
-   * @param {number} blockstreamDelay Number of blocks by which blockstream is behind the blockchain
    */
-  public async sync(augur: Augur<TBigNumber>, chunkSize: number, blockstreamDelay: number): Promise<void> {
-    let dbSyncPromises = [];
-    const highestAvailableBlockNumber = await augur.provider.getBlockNumber();
-    for (let dbIndex in this.syncableDatabases) {
-      dbSyncPromises.push(
-        this.syncableDatabases[dbIndex].sync(
-          augur,
-          chunkSize,
-          blockstreamDelay,
-          highestAvailableBlockNumber
-        )
-      );
-    }
+  public async sync(ethereumProvider: Provider, events: Events, chunkSize: number): Promise<void> {
+    const startSyncBlockNumber = await this.getSyncStartingBlock();
+    const endSyncBlockNumber = await ethereumProvider.getBlockNumber() - this.blockstreamDelay;
+    let firstBlockNumberInChunk = startSyncBlockNumber;
 
-    for (let trackedUser of await this.trackedUsers.getUsers()) {
+    while (firstBlockNumberInChunk < endSyncBlockNumber) {
```

Comment:
Should that be <= ?

Reply:
Yeah, good catch. I think it should be. Will change it.